File: kotlin/src/main/kotlin/model/BallPull.kt
package model

class BallPull(pull: String) {
    private val pull: String

    init {
        this.pull = pull
    }

    fun getRedBallsPulled(): Int {
        return splitTheBalls()["red"] ?: 0
    }

    fun getGreenBallsPulled(): Int {
        return splitTheBalls()["green"] ?: 0
    }

    fun getBlueBallsPulled(): Int {
        return splitTheBalls()["blue"] ?: 0
    }

    private fun splitTheBalls(): Map<String, Int> {
        var red = 0
        var green = 0
        var blue = 0

        val splitBalls = mutableMapOf<String, Int>()

        for (balls in pull.split(", ")) {
            val number = balls.split(" ")[0].toInt()
            val colour = balls.split(" ")[1]

            when (colour) {
                "red" -> {
                    red = number
                }
                "green" -> {
                    green = number
                }
                "blue" -> {
                    blue = number
                }
            }
        }

        splitBalls["red"] = red
        splitBalls["green"] = green
        splitBalls["blue"] = blue

        return splitBalls
    }
}

File: python/src/main/python/model/ball_pull.py
class BallPull:
    def __init__(self, pull: str) -> None:
        self.__pull = pull

    def get_red_balls_pulled(self) -> int:
        return self.__split_the_balls()["red"]

    def get_green_balls_pulled(self) -> int:
        return self.__split_the_balls()["green"]

    def get_blue_balls_pulled(self) -> int:
        return self.__split_the_balls()["blue"]

    def __split_the_balls(self) -> dict[str, int]:
        red: int = 0
        green: int = 0
        blue: int = 0

        split_balls: dict[str, int] = {}

        for balls in self.__pull.split(", "):
            number: int = int(balls.split(" ")[0])
            colour: str = balls.split(" ")[1]

            match colour:
                case "red":
                    red = number
                case "green":
                    green = number
                case "blue":
                    blue = number

        split_balls["red"] = red
        split_balls["green"] = green
        split_balls["blue"] = blue

        return split_balls
